core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystem.java

```diff
@@ -62,6 +62,9 @@ public interface RelDataTypeSystem {
   /** Returns the maximum precision of a NUMERIC or DECIMAL type. */
   int getMaxNumericPrecision();
 
+  /** Returns whether the scale of a NUMERIC or DECIMAL type can be negative. */
```

Comment:
The default value is 'false'. 
Because before this PR, when we execute like 
```
values cast(153 as decimal(3, -1));
```
will throw a parse exception. After we support the parse, the default behavior  will convert it to 
```
values cast(153 as decimal(3, 0));
```
So I set 'false' as the default value to keep the existing semantics unchanged.

Reply:
Based on my question below and other systems, maybe the right way to do this is to have a minimumNumericScale(), and this function to return 'true' if the minimum is less than 0.
You still have to have a rule to reject some types when the negative scale is too large.